### DISCUSSION TITLE
BackgroundTasks not working in arm64 such as AWS graviton chip

### DISCUSSION THREAD

**Discussion by jedihugo** [fastapi/fastapi#8678] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def function_name(
    request: Request,
    background_tasks: BackgroundTasks,
):
    background_tasks.add_task(
        function_call, param1, param2
    )
```

### Description

Recently switched from m5/r5 EC2 to arm64 chip AWS graviton m6/r6 EC2. And all the BackgroundTasks no longer running..

### Operating System

Linux

### Operating System Details

Ubuntu 20.04, running on AWS graviton 2 chip

### FastAPI Version

0.70.0

### Python Version

3.8

### Additional Context

BackgroundTasks is working fine in x86 machine. So I suspect it has to be the chip...

**Comment by oligond** [comment]
@jedihugo Two (+1) things you might consider to get feedback on this:

- The example code is currently not sufficient as it's incomplete and lacks context (e.g. the function called by the background task) - consequently, it is not useful for reproducing the issue (which might be hard to reproduce, at least requiring quite some effort, even if the example was complete)
- How about other context such as log files? Try to gather more information _why_ your background tasks are not running.
- Is it really an issue that you can hope to get help with here? According to your own assessment so far, some community focusing on AWS and/or hardware related issues might be a more promising starting point... 
   > I suspect it has to be the chip

**Comment by jedihugo** [comment]
@oligond Thanks for the comment. I understand my code is not sufficient but having a hard time how to create a reproduce-able example. I am also not sure how to access log files of background task, it just silently running in the background. Also not too expert of a programmer myself.

My test is simply:
- Comparing running simple `BackgroundTask` in m5 instance vs m6 instance (arm chip):
- Comparing running simple asyncio.get_event_loop().create_task() in m5 instance vs m6 instance (arm chip)
  ```python
  # In m5 ec2 instance (x86)
  query_bgtask = "INSERT INTO test VALUES ('m5_bgtask')"
  background_tasks.add_task(db.execute, query_bgtask)
  query_createtask = "INSERT INTO test VALUES ('m5_createtask')"
  asyncio.get_event_loop().create_task(db.execute(query_createtask))

  # In m6 ec2 instance (ARM)
  query_bgtask = "INSERT INTO test VALUES ('m6_bgtask')"
  background_tasks.add_task(db.execute, query_bgtask)
  query_createtask = "INSERT INTO test VALUES ('m6_createtask')"
  asyncio.get_event_loop().create_task(db.execute(query_createtask))
  ```
  - `asyncio` `create_task` works fine (inserts fine) in both m5 and m6 (arm)
  - `BackgroundTasks` does not insert the value `m6_bgtask`

I guess this is more of `**reporting potential issue**` with `arm64` chip, i.e. m6 ec2 instance, but I can tell you that `BackgroundTasks` works fine in x86 such as m5 ec2 instance. 

Regarding your last point, yes, I suspect it is chip problem, but considering `arm` chip is gaining momentum, it might be a good idea to look into some incompatibilities. 

One other example that I found (probably should report in separate issue), but might as well write it here while in topic of chip incompatibility:
In ARM chip such as M1 Mac and m6 ec2 instance, the `Body` example parameter is not working properly 
```python
from fastapi import APIRouter, Body
from typing import List, Dict
router = APIRouter()

async def function(
  request: Request,
  invoice_date: str = Body(..., example="2021-08-17"),
  customer_account_id: int = Body(..., example=123),
  supplier_id: int = Body(..., example=28),
  supplier_ref: str = Body(..., example="L337TZ"),
  order_data: List[Dict] = Body(..., example=[
    {
      "airline": "Delta Airlines",
      "ticket_no": "126-1234567890",
      "route_schedule": "JFK-LAX",
      "pax": "JOHN/DOE MR",
      "currency": "USD",
      "price": 1420.00, 
     },
  ])
):
```
| x86 chip  | Arm chip |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/4241302/141983903-44af7e51-0b21-4181-8eef-2901bf5c885d.png)  | !![image](https://user-images.githubusercontent.com/4241302/141984420-ced2e253-d8b0-478e-8ad6-45d52700e19c.png)  |

Feel free to close the issue as appropriate. I am just reporting potential incompatibility with arm chip.
